app/vminsert/netstorage/insert_ctx.go

```diff
@@ -135,6 +142,12 @@ func (ctx *InsertCtx) WriteDataPointExt(storageNodeIdx int, metricNameRaw []byte
 	br := &ctx.bufRowss[storageNodeIdx]
 	snb := ctx.snb
 	sn := snb.sns[storageNodeIdx]
+	if dropSamplesOlderThan.Milliseconds() > 0 {
+		if timestamp < int64(fasttime.UnixTimestamp()*1000)-(dropSamplesOlderThan.Milliseconds()) {
+			sn.rowsDroppedOnAge.Add(1)
```

Comment:
I don't think this metric must be at per-vmstorage level, since it doesn't depend on the particular vmstorage. It is better to make this metric global.